lua/cmake-tools/init.lua

```diff
@@ -352,38 +353,55 @@ function cmake.run(opt, callback)
   else -- if result_code == Types.SELECTED_LAUNCH_TARGET_NOT_BUILT
     -- Build select launch target every time
     config.build_target = config.launch_target
-    return cmake.build({ fargs = utils.deepcopy(opt.fargs) }, function()
-      vim.schedule(function()
-        result = config:get_launch_target()
-        local target_path = result.data
-        local new_s = vim.fn.fnamemodify(target_path, ":h")
-
-        if (osys.is_win32 == 1) then
-          return utils.execute(target_path, {
-            bufname = vim.fn.expand("%:p"),
-            cmake_launch_path = new_s,
-            cmake_console_position = const.cmake_console_position,
-            cmake_console_size = const.cmake_console_size,
-            cmake_launch_args = cmake:get_launch_args(),
-            cmake_unify_terminal_for_launch = const.cmake_unify_terminal_for_launch,
-            cmake_terminal_opts = const.cmake_terminal_opts
-          })
-        else
-          if not const.cmake_unify_terminal_for_launch then
-            target_path = '"' .. target_path .. '"'
+    if const.cmake_unify_terminal_for_launch then
+      -- DONOT schedule (or) schedule_wrap this piece of code
+      -- BUG: Without this if-else logic, the forwarding to terminal.execute() from utils.execute(), in utils.lua does not work.
```

Comment:
@Civitasv, please do not remove this if-else logic, as there is a bug on my machine. It does not call `terminal.execute()` from `utils.execute()`.

I will remove it later, but for now works. Otherwise, it launches in a quickfix list from `utils.run()` instead of `utils.execute()`...

Reply:
Indeed weird.